pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -199,6 +199,18 @@ func (c *Cluster) checkHealth() error {
 			continue
 		}
 
+		// don't failover if the mon pod is portable and not scheduled
+		if c.spec.Mon.VolumeClaimTemplate != nil {
```

Comment:
To satisfy #7776, I don't think we can postpone mon failover indefinitely. I'd rather ensure the mon failover happens, just with a longer timeout. I'm thinking:
- Per previous comment let's not treat the mons on a PVC special since we don't know if they're actually portable, so let's remove this line
- If the mon is not assigned to a node, let's only skip the failover once. So lines 208-210 should only be triggered once (effectively doubling the timeout). After the timeout happens the second time, we would still proceed with the mon failover.